### PR TITLE
[MINOR][SQL][DOCS] Improve spark.sql.files.minPartitionNum's doc

### PR DESCRIPTION
**Pull request by yikf**
### What changes were proposed in this pull request?

This PR aims to improve `spark.sql.files.minPartitionNum`'s doc.

### Why are the changes needed?

Improve description

### Does this PR introduce _any_ user-facing change?

Yes, better config description.

### How was this patch tested?

GA

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1740,8 +1740,10 @@ object SQLConf {
 
   val FILES_MIN_PARTITION_NUM = buildConf("spark.sql.files.minPartitionNum")
     .doc("The suggested (not guaranteed) minimum number of split file partitions. " +
-      "If not set, the default value is `spark.default.parallelism`. This configuration is " +
-      "effective only when using file-based sources such as Parquet, JSON and ORC.")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:1744]: I think you can just simply replace `spark.default.parallelism` to `spark.sql.leafNodeDefaultParallelism`. If users are interested in, they will read `spark.sql.leafNodeDefaultParallelism`

Comment (ulysses-you) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:1744]: +1

Comment (Yikf) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Thanks all guys, updated

**Comment by amaliujia** [comment]
Looks like the failed test is a flaky one so not relevant to this change.

**Comment by Yikf** [comment]
Yea, I can re-trigger if it's necessary.

**Comment by amaliujia** [comment]
I guess better to trigger to have it pass

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**